FileSystem/FileIO.js

```diff
@@ -0,0 +1,254 @@
+ */
+function closeFile(fileNumber) {
+  if (!(fileNumber in this.openFiles)) {
+    throw Error('File Number: ' + fileNumber + ' is not open');
+  }
+
+  var file = this.openFiles[fileNumber];
+
+  if (file.openMode == OpenMode.BINARY) {
+    DriveApp.getFileById(file.fileId).getBlob().setBytes(file.content); // Does this work??
+  } else {
+    DriveApp.getFileById(file.fileId).setContent(file.content);
+  }
+
+  delete this.openFiles[fileNumber];
+}
+
+/**
+ * Emulates VBA close statement API
+ * @param {Array.<number>} fileNumberList List of file numbers to close
+ */
+function closeFileList(fileNumberList) {
+  // Set default argument
+  fileNumberList = fileNumberList || [];
+
+  // Close all files
+  if (fileNumberList.length == 0) {
+    for (fileNumber in this.openFiles) {
+      this.closeFile(fileNumber);
+    }
+  }
+
+  // Close selected files
+  for (var i = 0; i < fileNumberList.length; i++) {
+    this.closeFile(fileNumberList[i]);
+  }
+}
+
+/**
+ * Emulates VBA freefile statement API
+ * If range = 0, return number from 1 to 255
+ * Else if range = 1, return number from 256 to 511
+ * @param {number} range File Number Range
+ */
+function getNextAvailableFile(range) {
+  // Set default argument
+  range = range || 0;
+
+  if (range == 0) {
+    for (var fileNumber = 1; fileNumber <= 255; fileNumber++) {
+      if (!(fileNumber in this.openFiles)) {
+        return fileNumber;
+      }
+    }
+  } else {
+    for (var fileNumber = 256; fileNumber <= 511; fileNumber++) {
+      if (!(fileNumber in this.openFiles)) {
+        return fileNumber;
+      }
+    }
+  }
+
+  throw Error('No free filenumber available');
+}
+
+/**
+ * Emulates VBA lof statement API
+ * @param {number} fileNumber File number
+ * @return {number} length of the open file
+ */
+function lof(fileNumber) {
+  if (!(fileNumber in this.openFiles)) {
+    throw Error('File Number: ' + fileNumber + ' is not open');
+  }
+  return this.openFiles[fileNumber].content.length;
+}
+
+/**
+ * Emulates VBA eof statement API
+ * @param {number} fileNumber File number
+ * @return {boolean} True if end of file is reached
+ */
+function isEOF(fileNumber) {
+  if (!(fileNumber in this.openFiles)) {
+    throw Error('File Number: ' + fileNumber + ' is not open');
+  }
+  var file = this.openFiles[fileNumber];
+  return file.content.length == file.pointer;
+}
+
+/**
+ * Emulates VBA print statement API
+ * @param {number} fileNumber File number
+ * @param {Array} outputList Expression List
+ */
+function printToFile(fileNumber, outputList) {
+  if (!(fileNumber in this.openFiles)) {
+    throw Error('File Number: ' + fileNumber + ' is not open');
+  }
+
+  // Set default argument
+  outputList = outputList || [];
+
+  var file = this.openFiles[fileNumber];
+
+  for (var i = 0; i < outputList.length; i++) {
+    var exp = outputList[i];
+    if (typeof exp === 'string') printString(file, exp);
+    else if (typeof exp === 'number') printNumber(file, exp);
```

Comment:
Where are these methods declared? printNumber, printString...?

Reply:
Do not see them in this CL.

Reply:
Ah because the files are being shared, this function came in this PR. But the code is actually in this PR: https://github.com/googleinterns/filesystem-in-appsscript/pull/9. This happened because I was using AppScript as source of control. From now on I will use git as the version control and this won't happen in the future.

Reply:
Update: I removed it from this PR